pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4209 @@
+                        iname.classname, namespace, iname))
+
+        if not instance_store.exists(iname):
+            raise CIMError(
+                CIM_ERR_NOT_FOUND,
+                _format("Instance {0!A} not found in CIM repository namespace "
+                        "{1!A}", iname, namespace))
+
+        # Handle namespace deletion, currently hard coded.
+        # Issue #2062 TODO/AM 8/18 Generalize the hard coded handling into
+        # provider concept
+        classname_lower = iname.classname.lower()
+        if classname_lower == 'pg_namespace':
+            ns_classname = 'PG_Namespace'
+        elif classname_lower == 'cim_namespace':
+            ns_classname = 'CIM_Namespace'
+        else:
+            ns_classname = None
+        if ns_classname:
+            namespace = iname.keybindings['Name']
+
+            # Reflect the namespace deletion in the CIM repository
+            # This will check the namespace for being empty.
+            self.remove_namespace(namespace)
+
+        # Delete the instance from the CIM repository
+        instance_store.delete(iname)
+
+    ####################################################################
+    #
+    #   EnumerateInstances server request
+    #
+    ####################################################################
+
+    def EnumerateInstances(self, namespace, ClassName, LocalOnly=None,
+                           DeepInheritance=None, IncludeQualifiers=None,
+                           IncludeClassOrigin=None, PropertyList=None):
+        # pylint: disable=line-too-long
+        """
+        Implements a WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.EnumerateInstances`.
+
+        Gets a list of subclasses of `ClassName` or of the class hiearchy root
+        in the CIM repository then executes get instance for each to create
+        the list of instances to be returned.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            The name of the CIM namespace in the CIM repository (case
+            insensitive). Must not be `None`. Leading or trailing slash
+            characters are ignored.
+
+          ClassName (:term:`string`):
+            Name of the class for which instances are to be enumerated (case
+            independent).
+
+          LocalOnly (:class:`py:bool`):
+            Controls the exclusion of inherited properties from the returned
+            instances.
+
+            This parameter has been deprecated in :term:`DSP0200` version 1.2
+            and should be set to `False` by the caller to avoid confusion.
+
+            This responder ignores the value of this parameter and acts as if
+            the parameter is set `False`, returning all properties include
+            inherited properties since that was the only way to
+            get around issues between different versions of :term:`DSP0200` that
+            defined incompatible behaviors to this request parameter.
```

Comment:
Some issues with these 3 paragraphs:
* Since this provider ignores it, this parameter does not control the exclusion.
* The information in the second paragraph is somewhat disconnected.
* I don't think the third paragraph should detail different versions of DSP0200.

How about replacing these 3 paragraphs with the following:
```
This parameter used to exclude inherited properties in the returned instances.
:term:`DSP0200` version 1.2 has deprecated this parameter and allows servers
to honor it or ignore it, treating it as True or False, as long as that is done consistently.

This provider ignores the parameter and includes inherited properties
in the returned instances.
```

Reply:
Modified both this and GetInstance